flashinfer/cute_dsl_attention.py

```
@@ -0,0 +1,283 @@
+
+
+def _load_fa_module(module_path: str):
+    """Load the flash attention module from a file path."""
+    spec = importlib.util.spec_from_file_location("flash_attention_v2", module_path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
+@functools.lru_cache(maxsize=32)
+def _get_compiled_kernel(head_dim, m_block, n_block, num_threads, is_causal, dtype_str):
```

Ruff (0.15.1)

[warning] Unused function argument: `dtype_str`

(ARG001)